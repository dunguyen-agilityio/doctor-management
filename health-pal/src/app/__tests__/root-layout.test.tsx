import { render } from '@utils-test'

import RootLayout from '../_layout'

jest.mock('@/services/auth', () => ({
  getProfile: jest.fn(() => Promise.resolve(null)),
}))

jest.mock('expo-splash-screen', () => ({
  preventAutoHideAsync: jest.fn(),
  hideAsync: jest.fn(),
  setOptions: jest.fn(),
  hide: jest.fn(),
}))

jest.mock('expo-router', () => {
  return {
    useRouter: () => ({
      push: jest.fn(),
      replace: jest.fn(),
      back: jest.fn(),
    }),
    usePathname: jest.fn().mockReturnValue('/'),
    useLocalSearchParams: () => ({}),
    Stack: ({ children }: { children: React.ReactNode }) => children,
    Slot: ({ children }: { children: React.ReactNode }) => children,
  }
})

describe('<RootLayout />', () => {
  it('matches snapshot after loading', async () => {
    const { toJSON } = render(<RootLayout />)

    expect(toJSON()).toMatchSnapshot()
  })
})
