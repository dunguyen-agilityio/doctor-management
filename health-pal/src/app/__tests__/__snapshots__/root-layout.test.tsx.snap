// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RootLayout /> matches snapshot after loading 1`] = `
<View
  aria-label="Notifications (F8)"
  pointerEvents="box-none"
  role="region"
  style={
    {
      "bottom": 0,
      "flexDirection": "column",
      "left": 0,
      "maxWidth": "100%",
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 100000,
    }
  }
  tabIndex={-1}
>
  <View
    focusable={false}
    pointerEvents="box-none"
    style={
      {
        "flexDirection": "column",
        "maxWidth": "100%",
        "position": "absolute",
      }
    }
  />
</View>
`;
